Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+
+* Primitive Type : `int`, `short`, `char`, `long`, `boolean`, `byte`, `float`, `double`
+* Wrapper Type : `Integer`, `Short`, `Char`, `Long`, `Boolean`, `Byte`, `Float`, `Double`
+* Object Type: `String`, `StringBuilder`, `Calendar`, `ArrayList` etc.
+
+We made a distinction between **Wrapper Type** and general **Object Type** for a reason - wrapper types are closely linked with their primitive counterparts via [autoboxing and unboxing](https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html).
+
+Typically you can declare variables using the following syntax :
+
+```java
+//Primitive Data Type
+
+int i = 10;
+
+// Object Data Type
+//initiates an Float object with value 1.0
+// variable myFloat now points to the object
+Float myFloat = new Float(1.0);
+```
+
+But there are more to these variables, [read about them here](#TODO).
+
+### Classes & Objects
+
+Classes are groups of variables and operations on them. A class can have variables, methods (or functions) and constructors (or methods which are used to initiate, more on that later!).
+
+Think of a `Class` as a blueprint for creating something concrete. A `Class` tells you the 'what' and 'how' an `object` of that Class will look like once `instantiated`. In essence, it defines `properties` (say color, engine capacity) and `behavior` (stop, speed up, change gears, honk etc.) for a Car in this case. 
+
+Objects are _instances_ of a class. All objects are instances of a certain class. Imagine a class being a "template", which every Object copies to. When you create an Object, basically it creates a new object on the blueprint of a class. Now lets look at this from a little piece of code :
+
+```java
+// Car class
+public class Car {
+  // car name
+  private String name;
+	// car mannufacturer name
+	private String manufacturerName;
+  // Constructor
+	public Car(String name, String man) {
+		this.name = name;
+		this.manufacturerName = man;
+	}
+  // Getter method
+	public String getName() {
+		return name;
+	}
+  // Getter method
+	public String getManufacturerName() {
+		return manufacturerName;
+	}
+}
+
+Car modelS = new Car("Model S","Tesla");
+
+System.out.println("Full Car Name = " + modelS.getManufacturerName() + " " + modelS.getName());
+// prints Tesla Model S
+```
+
+So, `Car` is a Class, which has the fields or properties name and manufacturerName. `modelS` is an object of `Car` Class. So `modelS` also has the same properties and methods.
+
+### Access Modifiers
+
+Ok, so if `modelS` has all the same properties, I should be able to access `name` or `manufacturerName` right?
+
+```java
+System.out.println(modelS.name);
+```
+Results in :
+
+```asciidoc
+Main.java:13: error: name has private access in Car
+		System.out.println(modelS.name);
+		                         ^
+1 error
+```
+
+Why? Notice we had mentioned `private` before the variable `name` in class `Car`. So since its private, we cant touch it from outside!
```

Comment:
> So since its private, we cant touch it from outside!

Clearly mark boundaries. _outside_ here, in this context, means that you cannot access its value or alter its value from another class, even if you have instantiated the former class as an object there.

Also, no need for

> So since

Just using one of them is enough.